db/post_migrate/20220531105951_drop_user_chat_channel_last_reads.rb

```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+class DropUserChatChannelLastReads < ActiveRecord::Migration[7.0]
+  def up
+    # usage has been dropped in https://github.com/discourse/discourse-chat/commit/1c110b71b28411dc7ac3ab9e3950e0bbf38d7970
+    # but apparently table never got dropped
+    drop_table :user_chat_channel_last_reads
```

Comment:
Does this work? I suspect you'll need to add a `DROPPED_TABLES` constant like https://github.com/discourse/discourse/blob/main/spec/fixtures/db/post_migrate/drop_table/20990309014013_drop_email_logs_table.rb

Reply:
I saw we were doing this at lots of places in core 🤷‍♂️

Reply:
I think we should definitely be using `TableDropper` as per https://github.com/discourse/discourse/commit/6a3c8fe69c16ad7360046f145db6689c18e91005

I am sure I have gotten pulled up on this by automated rules, it should be stopped by https://github.com/discourse/discourse/blob/9ac85d61632dae6020a627e02aa40c93e9b5086a/lib/migration/safe_migrate.rb so I think maybe this would break after merge when we run the plugin tests?

So yeah should follow the pattern here https://github.com/discourse/discourse/blob/9ac85d61632dae6020a627e02aa40c93e9b5086a/db/migrate/20201110110952_drop_github_user_infos.rb . Maybe the places doing it in core were done before this rule was added?

Reply:
Maybe the rails 7 change possibly messed something up with our freedom patch? Since it should be stopping this https://github.com/discourse/discourse/blob/9ac85d61632dae6020a627e02aa40c93e9b5086a/lib/freedom_patches/safe_migrations.rb, see my other comment too, for some reason it did not get added inline here.